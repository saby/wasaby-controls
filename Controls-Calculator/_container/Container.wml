<ws:partial template="{{_options.content}}"
            attr:class="controls_Calculator_theme-{{_options.theme}}"
            scope="{{_options}}"
            on:CalculatorSetNumber="_setNumber()"
            on:CalculatorRegisterNumber="_registerNumber()"
            on:CalculatorUnregisterNumber="_unregisterNumber()"
            on:CalculatorRegisterCalculator="_registerCalculator()"
            on:CalculatorUnregisterCalculator="_unregisterCalculator()"/>
