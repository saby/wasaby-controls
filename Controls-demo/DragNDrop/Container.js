define('Controls-demo/DragNDrop/Container', [
   'UI/Base',
   'wml!Controls-demo/DragNDrop/Container/Container'
], function (Base, template) {
   'use strict';

   var ModuleClass = Base.Control.extend({
      _template: template,
      _hasList: true,
      _hasGrid: false,
      _hasTree: false,
      _hasNotes: false,
      _hasMasterDetail: false,
      _selectedKeys: []
   });
   ModuleClass._styles = ['Controls-demo/DragNDrop/Container/Container'];

   return ModuleClass;
});
