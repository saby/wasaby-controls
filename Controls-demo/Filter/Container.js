/**
 * Created by am.gerasimov on 17.07.2018.
 */
define('Controls-demo/Filter/Container', [
   'UI/Base',
   'wml!Controls-demo/Filter/Container',
   'Types/source',
   'Controls-demo/Utils/MemorySourceData',
   'Controls/search',
   'Controls/filter'
], function (Base, template, sourceLib, memorySourceData) {
   'use strict';
   var SearchContainer = Base.Control.extend({
      _template: template,
      _navigation: null,
      _filter: null,
      _filterTabs: null,
      _filterButtonData: null,

      _beforeMount: function () {
         this._navigation = {
            source: 'page',
            view: 'page',
            sourceConfig: {
               pageSize: 20,
               page: 0,
               hasMore: false
            }
         };
         this._filter = {};
         this._filterTabs = {};
         this._filterButtonData = [
            {
               name: 'department',
               resetValue: 'По департаменту',
               value: 'По департаменту',
               viewMode: 'frequent',
               editorOptions: {
                  keyProperty: 'title',
                  displayProperty: 'title',
                  source: new sourceLib.Memory({
                     data: [
                        { id: 0, title: 'По департаменту' },
                        { id: 1, title: 'Разработка' },
                        { id: 2, title: 'Продвижение СБИС' },
                        { id: 3, title: 'Федеральная клиентская служка' },
                        { id: 4, title: 'Служба эксплуатации' },
                        { id: 5, title: 'Технологии и маркетинг' },
                        {
                           id: 6,
                           title: 'Федеральный центр продаж. Call-центр Ярославль'
                        },
                        { id: 7, title: 'Сопровождение информационных систем' }
                     ]
                  })
               }
            },
            {
               name: 'owner',
               resetValue: '0',
               value: '0',
               viewMode: 'frequent',
               editorOptions: {
                  keyProperty: 'owner',
                  displayProperty: 'title',
                  source: new sourceLib.Memory({
                     data: [
                        { id: 0, title: 'По ответственному', owner: '0' },
                        { id: 1, title: 'Новиков Д.В.', owner: 'Новиков Д.В.' },
                        { id: 2, title: 'Кошелев А.Е.', owner: 'Кошелев А.Е.' },
                        {
                           id: 3,
                           title: 'Субботин А.В.',
                           owner: 'Субботин А.В.'
                        },
                        {
                           id: 4,
                           title: 'Чеперегин А.С.',
                           owner: 'Чеперегин А.С.'
                        }
                     ]
                  })
               }
            },
            {
               name: 'owner',
               resetValue: '0',
               value: '0',
               source: new sourceLib.Memory({
                  data: [
                     { id: '0', title: 'По ответственному', lastName: '0' },
                     {
                        id: '1',
                        title: 'Новиков Д.В.',
                        lastName: 'Новиков Д.В.'
                     },
                     {
                        id: '2',
                        title: 'Кошелев А.Е.',
                        lastName: 'Кошелев А.Е.'
                     },
                     {
                        id: '3',
                        title: 'Субботин А.В.',
                        lastName: 'Субботин А.В.'
                     },
                     {
                        id: '4',
                        title: 'Чеперегин А.С.',
                        lastName: 'Чеперегин А.С.'
                     }
                  ],
                  keyProperty: 'id'
               })
            }
         ];
         this._source = new sourceLib.Memory({
            data: memorySourceData.departments,
            keyProperty: 'id'
         });
      }
   });

   SearchContainer._styles = ['Controls-demo/Filter/Container'];

   return SearchContainer;
});
