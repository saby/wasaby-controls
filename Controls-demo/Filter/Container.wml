<div class="controls-demo__filter-container">
    <Controls.browser:Browser bind:filter="_filter" filterButtonItems="{{_filterButtonData}}"
                              source="{{_source}}" navigation="{{_navigation}}" keyProperty="id">
        <div>
            <div class="controls-demo__filter-filtersWrapper">
                <Controls.filter:ViewContainer>
                    <Controls.filter:View attr:class="controls-demo__filterButton" detailPanelTemplateName="Controls-demo/Filter/Panel"/>
                </Controls.filter:ViewContainer>
            </div>
            <Controls.list:Container>
                <Controls.list:View emptyTemplate="Не найдено">
                    <ws:itemTemplate>
                        <ws:partial template="Controls/list:ItemTemplate">
                            <ws:contentTemplate>
                                <div style="padding: 4px; display: flex; justify-content: space-between">
                                    <div style="width: 300px">{{itemTemplate.item.contents.get('department')}}</div>
                                    <div style="font-size: 13px; color: darkblue">{{itemTemplate.item.contents.get('owner')}}(78)</div>
                                    <div style="font-size: 13px; color: limegreen">{{itemTemplate.item.contents.get('id')}}</div>
                                </div>
                            </ws:contentTemplate>
                        </ws:partial>
                    </ws:itemTemplate>
                </Controls.list:View>
            </Controls.list:Container>
        </div>
    </Controls.browser:Browser>
</div>
