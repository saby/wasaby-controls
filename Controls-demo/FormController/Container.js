define('Controls-demo/FormController/Container', [
   'Env/Env',
   'UI/Base',
   'wml!Controls-demo/FormController/Container'
], function (Env, Base, tmpl) {
   'use strict';

   var module = Base.Control.extend({
      _template: tmpl,
      _afterUpdate: function (cfg) {
         Env.IoC.resolve('ILogger').info(cfg.record);
      }
   });

   return module;
});
