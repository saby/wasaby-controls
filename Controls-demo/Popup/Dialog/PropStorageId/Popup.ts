import { Control, IControlOptions, TemplateFunction } from 'UI/Base';
import controlTemplate = require('wml!Controls-demo/Popup/Dialog/PropStorageId/Popup');

export default class Popup extends Control<IControlOptions> {
    protected _template: TemplateFunction = controlTemplate;
}
