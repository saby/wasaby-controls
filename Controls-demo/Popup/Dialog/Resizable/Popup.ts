import { Control, IControlOptions, TemplateFunction } from 'UI/Base';
import controlTemplate = require('wml!Controls-demo/Popup/Dialog/Resizable/Popup');

class Popup extends Control<IControlOptions> {
    protected _template: TemplateFunction = controlTemplate;
}

export default Popup;
