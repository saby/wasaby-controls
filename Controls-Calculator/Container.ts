import { Control, TemplateFunction } from 'UI/Base';
import * as template from 'wml!Controls-Calculator/_container/Container';
import Calculator from './View';
import Number from './Number';
import { SyntheticEvent } from 'UICommon/Events';

/**
 * Контрол, связывающий числа вводимые в калькулятор и сам калькулятор.
 * @extends UI/Base:Control
 * @control
 * @demo Controls-Calculator-demo/ClickOnNumber/ClickOnNumber
 * @public
 */
export default class Container extends Control {
    protected _template: TemplateFunction = template;
    private _calculators: Record<string, Calculator> = {};
    private _numbers: Record<string, Number> = {};

    private _registerCalculator(_: SyntheticEvent, instance: Calculator, id: string): void {
        this._calculators[id] = instance;

        Object.values(this._numbers).forEach((item) => {
            return item.setCalculatorRegistered(true);
        });
    }

    private _unregisterCalculator(_: SyntheticEvent, id: string): void {
        delete this._calculators[id];

        if (!Object.keys(this._calculators).length) {
            Object.values(this._numbers).forEach((item) => {
                return item.setCalculatorRegistered(false);
            });
        }
    }

    private _registerNumber(_: SyntheticEvent, instance: Number, id: string): void {
        this._numbers[id] = instance;

        if (Object.keys(this._calculators).length) {
            instance.setCalculatorRegistered(true);
        }
    }

    private _unregisterNumber(_: SyntheticEvent, id: string): void {
        delete this._numbers[id];
    }
}
