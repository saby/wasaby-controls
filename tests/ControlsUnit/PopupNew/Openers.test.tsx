/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode, render } from 'react-dom';
import { ContainerForTest } from 'Controls/_popup/Popup/Container';
import { StackOpener, DialogOpener } from 'Controls/popup';
import { Controller as PopupController } from 'Controls/popupTemplateStrategy';
import { act } from 'react-dom/test-utils';

function PopupTemplate() {
    return <div></div>;
}

// TODO:
PopupTemplate.isReact = true;

describe('Controls/popup', () => {
    describe('Openers', () => {
        let container = null;
        let Controller;

        const initContainer = async () => {
            let control;
            await act(async () => {
                render(
                    <ContainerForTest
                        ref={(instance) => {
                            control = instance;
                        }}
                    />,
                    container
                );
            });
            let afterMountPromiseResolver;
            let afterMountPromise = new Promise((resolve) => {
                afterMountPromiseResolver = resolve;
            });
            const controlAfterMount = control._afterMount.bind(control);
            control._afterMount = () => {
                controlAfterMount();
                afterMountPromiseResolver();
            };

            if (control._mounted) {
                afterMountPromise = Promise.resolve();
            }

            await afterMountPromise;
        };

        beforeEach(async () => {
            Controller = new PopupController();
            Controller.init();
            jest.spyOn(Controller, '_logTemplateName').mockImplementation();
            container = document.createElement('div');
            document.body.appendChild(container);
            await initContainer();
        });
        afterEach(() => {
            Controller.destroy();
            Controller = null;
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        });

        test('Dialog: should open and close popup', async () => {
            await new Promise(async (resolve) => {
                const opener = new DialogOpener();
                await opener.open({
                    template: PopupTemplate,
                    // Передадим id сами, для того чтобы снепшот корректно работал
                    id: 'popup-123',
                    eventHandlers: {
                        onClose: () => {
                            expect(container.childNodes[0].childElementCount).toBe(0);
                            resolve();
                        },
                    },
                });
                expect(container.childNodes[0].childElementCount).toBe(1);
                expect(container).toMatchSnapshot();
                opener.close();
            });
        });

        test('Stack: should open and close popup', async () => {
            await new Promise(async (resolve) => {
                const opener = new StackOpener();
                await opener.open({
                    template: PopupTemplate,
                    // Передадим id сами, для того чтобы снепшот корректно работал
                    id: 'popup-123',
                    eventHandlers: {
                        onClose: () => {
                            expect(container.childNodes[0].childElementCount).toBe(0);
                            resolve();
                        },
                    },
                });
                expect(container.childNodes[0].childElementCount).toBe(1);
                expect(container).toMatchSnapshot();
                opener.close();
            });
        });

        test('should return correct isOpened state', async () => {
            await new Promise(async (resolve) => {
                const opener = new StackOpener();
                expect(opener.isOpened()).toBeFalsy();
                await opener.open({
                    template: PopupTemplate,
                    eventHandlers: {
                        onClose: () => {
                            expect(opener.isOpened()).toBeFalsy();
                            resolve();
                        },
                    },
                });
                expect(opener.isOpened()).toBeTruthy();
                opener.close();
            });
        });
    });
});
