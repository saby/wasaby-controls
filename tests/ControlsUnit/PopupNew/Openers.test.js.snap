// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls/popup Openers Dialog: should open and close popup 1`] = `
<div>
  <div
    class="controls-PopupContainer ws-float-area-stack-panel-overflow"
    id="popup"
    tabindex="0"
  >
    <div
      class="controls-Popup ws-float-area-show-complete controls_themes__wrapper controls-Scroll_webkitOverflowScrollingTouch controls-Popup__lastItem"
      popupkey="popup-123"
      style="position: absolute; left: -10000px; top: -10000px; z-index: 10; height: auto; pointer-events: none;"
      tabindex="0"
    >
      <div />
    </div>
  </div>
</div>
`;

exports[`Controls/popup Openers Stack: should open and close popup 1`] = `
<div>
  <div
    class="controls-PopupContainer ws-float-area-stack-panel-overflow"
    id="popup"
    tabindex="0"
  >
    <div
      class="controls-Popup ws-float-area-show-complete  controls-Stack__last-item controls-Popup_shown controls_themes__wrapper controls-Scroll_webkitOverflowScrollingTouch controls-Popup__lastItem"
      popupkey="popup-123"
      style="position: fixed; bottom: 0px; z-index: 10; height: 768px; pointer-events: all;"
      tabindex="0"
    >
      <div
        class="controls_popupTemplate_theme-default controls-Stack__content-wrapper ws-workspace-width-NaN controls_popupTemplate_theme-default controls-Popup__template controls-Popup__template-height"
        tabindex="0"
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
